new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Application.java

```diff
@@ -78,6 +80,42 @@ public static boolean appAccessibleInPod(
       getLogger().warning(String.format("Failed to find pod %s to check the app", podName));
       return false;
     }
-  } 
+  }
+
+  /**
+   * Check if an application is accessible inside a WebLogic server pod.
+   *
+   * @param namespace Kubernetes namespace where the WebLogic server pod is running
+   * @param podName name of the WebLogic server pod
+   * @param port internal port of the managed server running in the pod
+   * @param appPath path to access the application
+   * @param expectedResponse expected response from the app
+   * @return true if the command succeeds
+   */
+  public static boolean appAccessibleInPodKubectl(
+      String namespace,
+      String podName,
+      String port,
+      String appPath,
+      String expectedResponse
+  ) {
+
+    // calling "kubectl exec" command to access the app inside a pod
+    String cmd = String.format(
+        "kubectl -n %s exec -it %s -- /bin/bash -c 'curl http://%s:%s/%s'",
+        namespace,
+        podName,
+        podName,
+        port,
+        appPath);
+
+    CommandParams params = Command
+        .defaultCommandParams()
+        .command(cmd)
+        .saveResults(true)
+        .redirect(false)
+        .verbose(false);
+    return Command.withParams(params).executeAndVerify(expectedResponse);
```

Comment:
For other tests, we've needed to have multiple retries for initial curl attempts as tests can sometimes try access an app before it or the service is fully deployed; in addition, it's been helpful to tune the curl command's connection and overall timeouts.

Reply:
For MII sample testing, we use "--connect-timeout 5 --max-time 20" and loop the curl command up to 15 times on a failure (5 seconds between each try).

Reply:
The test that uses this assertion has a retry logic at the higher level; this is the pattern used in the Juni5 integration tests.